src/main/java/teammates/common/util/Config.java

```diff
@@ -212,21 +267,7 @@ public static boolean isDevServerLoginEnabled() {
      * {@code relativeUrl} must start with a "/".
      */
     public static AppUrl getFrontEndAppUrl(String relativeUrl) {
-        if (Config.isDevServer() && APP_FRONTENDDEV_URL != null) {
-            return new AppUrl(APP_FRONTENDDEV_URL + relativeUrl);
-        }
-
-        // In production, the back-end and front-end lives under the same domain
-        return getBackEndAppUrl(relativeUrl);
```

Comment:
I don't think this behavior should be removed (not the scope of this issue anyway). You're likely right that `getBackEndAppUrl` can be removed, but what should be done here is replacing this with `new AppUrl(getBaseAppUrl() + relativeUrl)` directly.
Remember that `APP_FRONTEND_URL` is an optional property.